ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpDataHelper.java

```diff
@@ -0,0 +1,630 @@
+   * @param putRecord {@code true} if the record is a Put record, {@code false} otherwise (incase of a Delete record)
+   * @param isExpired {@code true} if the blob has expired, {@code false} otherwise
+   */
+  private void updateBlobIdToLogRecordMap(ConcurrentHashMap<String, LogBlobRecord> blobIdToLogRecord, String blobId,
+      Long offset, boolean putRecord, boolean isExpired) {
+    if (blobIdToLogRecord != null) {
+      if (blobIdToLogRecord.containsKey(blobId)) {
+        if (putRecord) {
+          blobIdToLogRecord.get(blobId).addPutRecord(offset, isExpired);
+        } else {
+          blobIdToLogRecord.get(blobId).addDeleteRecord(offset);
+        }
+      } else {
+        blobIdToLogRecord.put(blobId, new LogBlobRecord(offset, putRecord, isExpired));
+      }
+    }
+  }
+
+  /**
+   * Returns if the blob has been expired or not based on the time to live value
+   * @param timeToLive time in milliseconds referring to the time to live for the blob
+   * @return {@code true} if blob has expired, {@code false} otherwise
+   */
+  private boolean isExpired(Long timeToLive) {
+    return timeToLive != Utils.Infinite_Time && SystemTime.getInstance().milliseconds() > timeToLive;
+  }
+
+  /**
+   * Log a message to the standard out or to the file as per configs
+   * @param msg the message that needs to be logged
+   */
+  synchronized void logOutput(String msg, LogLevel logLevel) {
+    if (logLevel != LogLevel.DEBUG || (!avoidTraceLogging)) {
+      try {
+        if (fileWriter == null) {
+          System.out.println(msg);
+        } else {
+          fileWriter.write(msg + "\n");
+        }
+      } catch (IOException e) {
+        System.out.println("IOException while trying to write to File");
+      }
+    }
+  }
+
+  /**
+   * Returns the {@link ClusterMap}
+   * @return the {@link ClusterMap}
+   */
+  ClusterMap getClusterMap() {
+    return this._clusterMap;
+  }
+
+  /**
+   * Flushes and closes the outfile if need be
+   */
+  void shutdown() {
+    try {
+      if (fileWriter != null) {
+        fileWriter.flush();
+        fileWriter.close();
+      }
+    } catch (IOException IOException) {
+      System.out.println("IOException while trying to close File " + outFile);
+    }
+  }
+
+  /**
+   * Holds the information about a single record in the index
+   */
+  class IndexRecord {
+    String message;
+    boolean isDeleted;
+    boolean isExpired;
+
+    public IndexRecord(String msg, boolean isDeleted, boolean isExpired) {
+      this.message = msg;
+      this.isDeleted = isDeleted;
+      this.isExpired = isExpired;
+    }
+
+    String getMessage() {
+      return message;
+    }
+
+    boolean isDeleted() {
+      return isDeleted;
+    }
+
+    boolean isExpired() {
+      return isExpired;
+    }
+
+    public String toString() {
+      return message + ", isDeleted:" + isDeleted + ", isExpired:" + isExpired;
+    }
+  }
+
+  /**
+   * Holds information about a blob in the log
+   */
+  class LogBlobRecord {
```

Comment:
private

Also there are some public methods you may want to change to package level.